tests/unit/distribution/Gaussian_unittest.cc

```diff
@@ -0,0 +1,187 @@
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ * this
+ * list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+ * FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * The views and conclusions contained in the software and documentation are
+ * those
+ * of the authors and should not be interpreted as representing official
+ * policies,
+ * either expressed or implied, of the Shogun Development Team.
+ */
+
+#include <gtest/gtest.h>
+#include <shogun/base/some.h>
+#include <shogun/distributions/Gaussian.h>
+#include <shogun/features/DenseFeatures.h>
+#include <shogun/lib/SGMatrix.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/mathematics/Math.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+
+using namespace shogun;
+
+TEST(Gaussian, log_pdf_single_1d)
+{
+	SGVector<float64_t> mean(1), point(1);
+	SGMatrix<float64_t> cov(1, 1);
+
+	mean[0] = 0;
+	point[0] = 0;
+	cov(0,0) = 1.0;
+
+	auto gauss = some<CGaussian>(mean, cov, FULL);
+	float64_t log_pdf = gauss->compute_log_PDF(point);
+	EXPECT_NEAR(log_pdf, -0.91893853320467267, 1e-16);
+}
+
+TEST(Gaussian, log_pdf_multiple_independent_2d)
+{
+	SGVector<float64_t> mean(2);
+	SGMatrix<float64_t> cov(2, 2);
+
+	mean[0] = mean[1] = 0;
+	cov(0,0) = 1.0;
+	cov(0,1) = 0.0;
+	cov(1,0) = 0.0;
+	cov(1,1) = 1.0;
+
+	auto gauss = some<CGaussian>(mean, cov, FULL);
+	float64_t log_pdf = gauss->compute_log_PDF(mean);
+	EXPECT_NEAR(log_pdf, -1.8378770664093453, 1e-16);
+}
+
+TEST(Gaussian, log_pdf_multiple_2d)
+{
+	SGVector<float64_t> mean(2);
+	SGMatrix<float64_t> cov(2,2);
+
+	mean[0] = mean[1] = 0;
+
+	cov(0,0) = 4.0;
+	cov(0,1) = 1.0;
+	cov(1, 0) = 1.0;
+	cov(1,1) = 2.0;
+	auto gauss = some<CGaussian>(mean, cov, FULL);
+	float64_t log_pdf = gauss->compute_log_PDF(mean);
+	EXPECT_NEAR(log_pdf, -2.8108321409370021, 1e-16);
+}
+
+TEST(Gaussian, train_univariate)
+{
+	float64_t eps = 1e-8;
+	sg_rand->set_seed(1);
```

Comment:
why do you need a fixed seed when you test for the estimation? Should be ok everytime no?